### PR TITLE
STONK-7: Frontend > Convert To Path Aliases

### PR DESCRIPTION
**Pull request by Jenna59**
## Ticket

#8 

## Changes

Updated vite.config and tsconfig to create path aliases. Updated the files in the components, reducers, and actions directories with the new path aliases.

## Developer Checklist

- [ Y] I have verified my changes to confirm that it works
- [ Y] I have added the appropriate label to this pull request
- [ Y] I have linked the corresponding issue to this pull request
- [ Y] I have followed the pull request title pattern of `STONK-NUMBER: Domain > Issue Title`

### CODE REVIEW

File: frontend/tsconfig.json [hunk 1]
```diff
@@ -15,6 +15,19 @@
...
+    "baseUrl": ".",
+    "paths": {
+      "@/*": ["./src/*"],
+      "actions/*": ["./src/actions/*"],
+      "api": ["./src/api/*"],
+      "images": ["./assets/images/*"],
+      "components/*": ["./components/*"],
+      "constants/*": ["./constants/*"],
+      "contexts/*": ["/contexts/*"],
+    }
```
Comment (JackyTea) [inline frontend/tsconfig.json:28]: For scoping conventions use `@`, see https://docs.npmjs.com/cli/v10/using-npm/scope for reference.

Example:
```json
"baseUrl": ".",
"paths": {
  "@/app/*": ["src/pages/app/*"],
  "@/constants/*": ["src/constants/*"],
  "@/hooks/*": ["src/hooks/*"],
  "@/pages/*": ["src/pages/*"],
  "@/utils/*": ["src/utils/*"],
}
```

File: frontend/src/reducers/transactions.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { GET_ALL_TRANSACTIONS } from '../constants/actions';
+import { GET_ALL_TRANSACTIONS } from 'constants/actions.js';
```
Comment (JackyTea) [inline frontend/src/reducers/transactions.js:1]: Is the `.js` extension needed here?

File: frontend/tsconfig.json (hunk 1 above)
Comment (Jenna59) [reply frontend/tsconfig.json]: ` "baseUrl": ".",
    "paths": {
      "@/*": ["src/*"],
      "@/actions/*": ["src/actions/*"],
      "@/api/*": ["src/api/*"],
      "@/assets*": ["src/assets/*"],
      "@/components/*": ["src/components/*"],
      "@/constants/*": ["src/constants/*"],
      "@/contexts/*": ["src/contexts/*"],
    }`

File: frontend/src/reducers/transactions.js (hunk 2 above)
Comment (Jenna59) [reply frontend/src/reducers/transactions.js]: ``import { GET_ALL_TRANSACTIONS } from '@/constants/actions';``

**Review by JackyTea — APPROVED** [review]

**Pull request merged**